### PR TITLE
fix(charts): PSP replaced by PSA baseline

### PR DESCRIPTION
**Pull request by kaanyagci**
## Description

Closes FL-574

Pod Security Policy (PSP) replaced by Pod Security Admission `baseline`. It could be `restricted` but `kaniiko` needs to run as `root` user.

## Adds
- `substra/ca-cert-injector`: As the installing `openssl` package was violating the `runAsNonRoot` rule privacy context, a standalone Docker image created make those package(s) pre-installed.
- `securityContext` added on pod and container level for PSA baseline 

## Changes
- `redis` subchart version incremented

## Removes
- Pod security policy removed

Related to [FL-574](https://linear.app/owkin/issue/FL-574/replace-deprecated-podsecuritypolicy-in-chart)

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [x] documentation was updated

### CODE REVIEW

File: docker/ca-cert-injector/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM debian:bookworm-slim
```
Comment (oleobal) [inline docker/ca-cert-injector/Dockerfile:1]: Could this cause problem given the backend image is based on python which is based on ubuntu (.. I think)?

**Review by oleobal — CHANGES_REQUESTED** [review]
This PR is good, could you update the chart changelog (`charts/substra-backend/CHANGELOG`) et make the title into an imperative ("create" rather than "created")?

Otherwise I think we can merge this as soon as the release is done.

File: docker/ca-cert-injector/Dockerfile (hunk 1 above)
Comment (kaanyagci) [reply docker/ca-cert-injector/Dockerfile]: I have no idea but I changed to `ubuntu:mantic` to avoid any risks 😉

**Comment by Owlfred** [comment]
```
Usage: /e2e [options] [help]

/e2e may appear anywhere as long as it is on its own line

Options:
  --refs <value>                                         Extra refs (branch or tag) with format REPO=GIT_REF,REPO=GIT_REF.
  Supported repositories: hlf-k8s, orchestrator, substra-backend, substra-frontend, substra-tools, substrafl, substra, substra-tests, substra-ci.
  Example: /e2e --refs substra-backend=some_branch,orchestrator=some_tag (default: {})
  --tests-to-run, --tests <tests-to-run>                 Comma-separated list of tests to run. Valid options: sdk,substrafl,frontend or NONE. (default: "sdk")
  --benchmarks-to-run, --benchmarks <benchmarks-to-run>  Comma-separated list of workflows tests to run. Valid options: mnist, camelyon or NONE. (default: "NONE")
  --orchestrator-mode, --mode <orchestrator-mode>        Comma-separated list of orchestrator modes to run tests for. Valid options: standalone,distributed (default: "standalone,distributed")
  -h, --help                                             display help for command
```

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: CHANGELOG.md [hunk 2]
```diff
@@ -28,7 +28,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - New `SECRET_KEY` optional environment variable ([#671](https://github.com/Substra/substra-backend/pull/671))
 - `/api-token-auth/` and the associated tokens can now be disabled through the `EXPIRY_TOKEN_ENABLED` environment variable and `server.allowImplicitLogin` chart value ([#698](https://github.com/Substra/substra-backend/pull/698))
 - Tokens issued by `/api-token-auth/` can now be deleted like other API tokens, through a `DELETE` request on the `/active-api-tokens` endpoint ([#698](https://github.com/Substra/substra-backend/pull/698))
-
+- A standalone Docker image for updating Private CA Certificates. ([726](https://github.com/Substra/substra-backend/pull/726))
```
Comment (oleobal) [inline CHANGELOG.md:31]: (ultra nitpick)

```suggestion
- A standalone Docker image for updating Private CA Certificates. ([#726](https://github.com/Substra/substra-backend/pull/726))
```

File: charts/substra-backend/templates/deployment-registry-prepopulate.yaml [hunk 3]
```diff
@@ -20,12 +20,32 @@ spec:
...
+        securityContext: &container-security-context
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop: 
+              - ALL
         command: ['dockerize', '-wait', 'tcp://{{ $.Release.Name }}-docker-registry:5000']
       - name: kaniko
         image: {{ include "common.images.name" $.Values.kaniko.image }}
+        securityContext:
+          # TODO: Blocks PSA
```
Comment (oleobal) [inline charts/substra-backend/templates/deployment-registry-prepopulate.yaml:42]: Is this TODO still relevant?

File: charts/substra-backend/templates/deployment-scheduler-worker.yaml [hunk 4]
```diff
@@ -32,6 +32,9 @@ spec:
         fsGroup: {{ .Values.schedulerWorker.podSecurityContext.fsGroup }}
         runAsUser: {{ .Values.schedulerWorker.podSecurityContext.runAsUser }}
         runAsGroup: {{ .Values.schedulerWorker.podSecurityContext.runAsGroup }}
+        # runAsNonRoot: true
```
Comment (oleobal) [inline charts/substra-backend/templates/deployment-scheduler-worker.yaml:35]: Should we uncomment this or remove it?

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Review by guilhem-barthes — APPROVED** [review]

**Pull request merged**